src/racer/util.rs

```diff
@@ -70,6 +70,27 @@ pub fn symbol_matches(stype: SearchType, searchstr: &str, candidate: &str) -> bo
     }
 }
 
+// Try to valid if the given scope contains a valid closure arg scope.
+pub fn closure_valid_arg_scope(scope_src: &str) -> Option<(usize, usize, &str)> {
+    // Try to find the left and right pipe, if one or both are not present, this is not a valid
+    // closure definition
+    let left_pipe = if let Some(pos) = scope_src.find('|') { pos } else { return None; };
+    let rest_scope = &scope_src[left_pipe + 1..];
+    let right_pipe = if let Some(pos) = rest_scope.find('|') {
+        left_pipe + 1 + pos
+    } else {
+        return None;
+    };
+
+    let pipe_scope = &scope_src[left_pipe..right_pipe+1];
+
+    // If we found not forbidden characters --> no closure definition
+    let forbidden_chars = [ ";", "{", "}" ];
```

Comment:
Struct patterns are allowed in closure arguments; **[here's](https://play.rust-lang.org/?gist=c5ad305c6c473028507a1926ff225664&version=stable)** an example. Please either a) have that work, and add a test to demonstrate it, or b) add a note to this method's doc comment calling out that struct patterns in closure arguments aren't supported.